### DISCUSSION TITLE
Menambahkan caption untuk map di leaflet dan modifikasi legend

### DISCUSSION THREAD

**Discussion by Saupia** [teamalgoritma/community#417] [Q&A] [answered]
Saya telah membuat map namun saya pikir lebih interaktif jika: 
(1) Legend warna yang saya pilih tidak menampilkan caption nama warna (mis, "yellow") tetapi menampilkan skala (mis, buble kuning tetapi caption di legend "Shifting"). Bagaiman caranya?

```
egids_map <- leaflet() %>% 
  setView(lng=117, lat= -0.7, zoom=4 ) %>% 
  addProviderTiles("Esri.WorldImagery", group = "background 1") %>% 
  addTiles(options = providerTileOptions(noWrap = TRUE), group="background 2") %>% 
  addCircleMarkers(data = egids7, lng=~Longitude , lat=~Latitude, radius=8 , color="black",
                   fillColor="yellow", stroke = TRUE, fillOpacity = 0.8, group="yellow") %>% 
  addCircleMarkers(data = egids8, lng=~Longitude , lat=~Latitude, radius=8 , color="black",
                   fillColor="red", stroke = TRUE, fillOpacity = 0.8, group="red") %>% 
  addCircleMarkers(data = egids8b,lng=~Longitude , lat=~Latitude, radius=8 , color="black",
                   fillColor="grey", stroke = TRUE, fillOpacity = 0.8, group="grey") %>% 
  addCircleMarkers(data = egids9,lng=~Longitude , lat=~Latitude, radius=8 , color="black",
                   fillColor="black", stroke = TRUE, fillOpacity = 0.8, group="black") %>% 
  addLayersControl(overlayGroups = c("yellow","red", "grey", "black") , baseGroups = c("background 1", "background 2"), 
                   options = layersControlOptions(collapsed = FALSE)) 
egids_map
```
(2) Saya membuat peta, apakah memungkinkan untuk menampilkan nama bahasa ketika kursor diarahkan ke bubble?
![image](https://user-images.githubusercontent.com/114889462/198191853-927739eb-1fcd-4936-812b-103d97db67de.png)

**Comment by ranydwi** [comment, marked as answer]
Dear Pak Renhard @Saupia,

Untuk pertanyaan Bapak yang pertama cara untuk mengubah nama di legend silahkan Pak Renhard mengganti value pada `addLayersControl` dengan nilai yang bapak inginkan misal karena Bapak ingin yellow = Shifting maka bisa diganti dengan code berikut:
```
addLayersControl(overlayGroups = c("Shifting","red", "grey", "black") , baseGroups = c("background 1", "background 2"), 
                   options = layersControlOptions(collapsed = FALSE)) 
```

Kemudian untuk pertanyaan kedua, tentu bisa ditambahkan nama bahasa untuk ditampilkan dengan mendefinisikan nilai label yang akan dimunculkan dengan contoh seperti berikut:
```
popup.cont1  <- paste("<h2><b>", egids7$((Variabel_nama bahasa)), "</b></h2>")
```
Kemudian pada setiap parameter `addCircleMarkers` silahkan tambahkan parameter `label = popup.cont1`

Demikian yang dapat saya beritahukan. Semoga dapat membantu Bapak
Team Algoritma.
